modules/global/src/de/diedavids/cuba/dataimport/converter/MetaPropertyMatcher.groovy

```diff
@@ -3,24 +3,44 @@ package de.diedavids.cuba.dataimport.converter
 import com.haulmont.chile.core.model.MetaClass
 import com.haulmont.chile.core.model.MetaProperty
 import de.diedavids.cuba.dataimport.entity.AttributeType
+import de.diedavids.cuba.dataimport.entity.ImportAttributeMapper
 import org.apache.commons.lang.StringUtils
 import org.springframework.stereotype.Component
 
 @Component('ddcdi_MetaPropertyMatcher')
 class MetaPropertyMatcher {
 
 
-    String findEntityAttributeForColumn(String column, MetaClass selectedEntity) {
-
-        MetaProperty possibleProperty = findPropertyByColumn(selectedEntity, column)
-
-        String result = ''
-        if (possibleProperty && isSimpleDatatype(possibleProperty)) {
-                result = possibleProperty.name
+    ImportAttributeMapper findEntityAttributeForColumn(Integer index, String column, MetaClass selectedEntity) {
```

Comment:
i think you mixed up the responsibilities here a little bit. From the naming: `ImportAttributeMapperCreator` is for creating the mappers. `MetaPropertyMatcher` is for matching, not for creating. I think the logic belongs to the former one.
Ill change that and add some tests, because there are some errors.

Reply:
Now that you mentioned, I see, I should have done this is `ImportAttributeMapperCreator`, sorry about that.
I agree the complicity is a bit limit here, need to be crated outside this method